torch/nn/functional.py

```diff
@@ -1264,6 +1264,44 @@ def alpha_dropout(input: Tensor, p: float = 0.5, training: bool = False, inplace
     return _VF.alpha_dropout_(input, p, training) if inplace else _VF.alpha_dropout(input, p, training)
 
 
+def dropout1d(input: Tensor, p: float = 0.5, training: bool = True, inplace: bool = False) -> Tensor:
```

Comment:
should there be feature_dropout_nd function that factors out common code between 1d/2d/3d? Doesn't need to happen now

Reply:
flagged for later - this should definitely happen